student-store-ui/src/components/ProductCard/ProductCard.jsx

```diff
@@ -0,0 +1,88 @@
+import { Link } from "react-router-dom";
+import { useState } from "react";
+
+export default function ProductCard({
+  name,
+  id,
+  image,
+  price,
+  product,
+  shoppingCart,
+  setShoppingCart,
+}) {
+  function addToCart(product) {
+    const checkProduct = shoppingCart?.find((item) => item?.id === product.id);
+    if (checkProduct) {
+      const updatedCart = shoppingCart?.map((item) => {
+        if (item?.id === product.id) {
+          return { ...item, quantity: item.quantity + 1 };
+        }
+        return item;
+      });
+      setShoppingCart(updatedCart);
+      console.log(updatedCart);
+    } else {
+      // spread operator and what it can take in?
+      setShoppingCart([...shoppingCart, { ...product, quantity: 1 }]);
+    }
+  }
+
+  function subToCart(product) {
+    const checkProduct = shoppingCart?.find((item) => item?.id === product.id);
+    console.log("checkProd", checkProduct);
+    if (checkProduct) {
+      let updatedCart = [...shoppingCart];
+      for (let i = 0; i < updatedCart.length; i++) {
+        if (updatedCart[i]?.id === product.id) {
+          if (updatedCart[i].quantity > 1) {
+            updatedCart[i].quantity--;
+          } else {
```

Comment:
I recommend checking the exemplar code for how they handled this, but it should be similar to what you wrote on line 26